go-controller/pkg/ovn/libovsdbops/nb_global.go

```diff
@@ -0,0 +1,61 @@
+package libovsdbops
+
+import (
+	"context"
+	"fmt"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+// findNBGlobal returns the NBGlobal table entry
+func findNBGlobal(nbClient libovsdbclient.Client) (*nbdb.NBGlobal, error) {
+	nbGlobal := []nbdb.NBGlobal{}
+	ctx, cancel := context.WithTimeout(context.Background(), types.OVSDBTimeout)
+	defer cancel()
+	err := nbClient.List(ctx, &nbGlobal)
+	if err != nil {
+		return nil, fmt.Errorf("can't find Locial Switches err: %v", err)
+	}
+
+	if len(nbGlobal) == 0 {
+		return nil, libovsdbclient.ErrNotFound
+	}
+	// The nbGlobal table should only have one row
+	if len(nbGlobal) != 1 {
+		return nil, fmt.Errorf("multible NBGlobal rows found")
+	}
+
+	return &nbGlobal[0], nil
+}
+
+// SetNBGlobalOptions mutates the options on the NBGlobal table
+func SetNBGlobalOptions(nbClient libovsdbclient.Client, options map[string]string) error {
+	// find the nbGlobal table's UUID, since we don't have any other way to reliably look this row up otherwise
+	nbGlobal, err := findNBGlobal(nbClient)
+	if err != nil {
+		return err
+	}
+
+	for k, v := range options {
+		nbGlobal.Options[k] = v
+	}
+
+	// Update the options column in the nbGlobal entry
+	opModel := OperationModel{
+		Model: nbGlobal,
+		OnModelUpdates: []interface{}{
```

Comment:
Mutate or Update? Update will override any other options, which may be what you want... or not.

Reply:
Will Clarify this... Because NBGlobal is a bit funny we don't know the name or UUID of the row so we need to do a lookup or Model Predicate search on the `hostname: nodename` column